File: jasmine-2.5.2/spec/BonusSpec.js
describe("Bonus", function() {
  var bonus;

  beforeEach(function() {
    bonus = new Bonus();
  });
  it('initialises default settings', function() {
    expect(bonus.getStrikeOrSpare()).toEqual("");
    expect(bonus.getNextMultiplier()).toEqual(0);
    expect(bonus.getNextButOneMultiplier()).toEqual(0);
  });
  it('records strikes', function() {
    bonus.recordStrikeOrSpare(1);
    expect(bonus.getStrikeOrSpare()).toEqual("Strike");
    expect(bonus.getNextMultiplier()).toEqual(1);
    expect(bonus.getNextButOneMultiplier()).toEqual(1);
  });
  it('records spares', function() {
    bonus.recordStrikeOrSpare(2);
    expect(bonus.getStrikeOrSpare()).toEqual("Spare");
    expect(bonus.getNextMultiplier()).toEqual(1);
    expect(bonus.getNextButOneMultiplier()).toEqual(0);
  });
  it('resets used bonuses', function() {
    bonus.recordStrikeOrSpare(1);
    bonus.useBonuses();
    expect(bonus.getNextMultiplier()).toEqual(1);
    expect(bonus.getNextButOneMultiplier()).toEqual(0);
  });
});

File: spec/unit/bonus_spec.rb
require 'bonus'

describe Bonus do
  before(:each) do
    Bonus.reset
  end

  describe '.strikes' do
    it 'stores an array of strike bonuses' do
      expect(Bonus.strikes).to be_an(Array)
    end
  end

  describe '.spares' do
    it 'stores an array of spare bonuses' do
      expect(Bonus.spares).to be_an(Array)
    end
  end

  describe '.add_spare' do
    it 'adds a bonus to the spare array' do
      Bonus.add_spare(6)
      expect(Bonus.spares).to eq [6]
    end
  end

  describe '.add_strike' do
    it 'adds a bonus to the strike array' do
      Bonus.add_strike(5)
      expect(Bonus.strikes).to eq [5]
    end
  end

end